service.autosubs/default.py

```diff
@@ -0,0 +1,161 @@
+def isExcluded(movieFullPath):
+    if not movieFullPath:
+        return False
+
+    Debug("isExcluded(): Checking exclusion settings for '%s'." % movieFullPath)
+
+    if (movieFullPath.find("pvr://") > -1) and getSettingAsBool('ExcludeLiveTV'):
+        Debug("isExcluded(): Video is playing via Live TV, which is currently set as excluded location.")
+        return False
+
+    if (movieFullPath.find("http://") > -1 or movieFullPath.find("https://") > -1) and getSettingAsBool('ExcludeHTTP'):
+        Debug("isExcluded(): Video is playing via HTTP source, which is currently set as excluded location.")
+        return False
+
+    ExcludePath = getSetting('ExcludePath')
+    if ExcludePath and getSettingAsBool('ExcludePathOption'):
+        if (movieFullPath.find(ExcludePath) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 1." % ExcludePath)
+            return False
+
+    ExcludePath2 = getSetting('ExcludePath2')
+    if ExcludePath2 and getSettingAsBool('ExcludePathOption2'):
+        if (movieFullPath.find(ExcludePath2) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 2." % ExcludePath2)
+            return False
+
+    ExcludePath3 = getSetting('ExcludePath3')
+    if ExcludePath3 and getSettingAsBool('ExcludePathOption3'):
+        if (movieFullPath.find(ExcludePath3) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 3." % ExcludePath3)
+            return False
+
+    ExcludePath4 = getSetting('ExcludePath4')
+    if ExcludePath4 and getSettingAsBool('ExcludePathOption4'):
+        if (movieFullPath.find(ExcludePath4) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 4." % ExcludePath4)
+            return False
+
+    ExcludePath5 = getSetting('ExcludePath5')
+    if ExcludePath5 and getSettingAsBool('ExcludePathOption5'):
+        if (movieFullPath.find(ExcludePath5) > -1):
+            Debug("isExcluded(): Video is playing from '%s', which is currently set as excluded path 5." % ExcludePath5)
+            return False
+
+    return True
+
+
+class AutoSubsPlayer(xbmc.Player):
+    def __init__(self, *args, **kwargs):
+        xbmc.Player.__init__(self)
+        Debug("Initalized")
+        self.run = True
+
+    def onPlayBackStopped(self):
+        Debug("Stopped")
+        self.run = True
+
+    def onPlayBackEnded(self):
+        Debug("Ended")
+        self.run = True
+
+    def onPlayBackStarted(self):
+        if (xbmc.Player().isPlayingVideo()):
+            check_for_specific = (__addon__.getSetting('check_for_specific').lower() == 'true')
+            specific_language = (__addon__.getSetting('selected_language'))
+            specific_language = xbmc.convertLanguage(specific_language, xbmc.ISO_639_2)
+            try:
+                xbmc.sleep(3000)
```

Comment:
please use xbmc.monitor().waitforabort

Reply:
ok. There's still a lot that could be reafctored. Did not aim to do that :) Just make it awailable for Matrix users :)